### ISSUE TITLE
How to dump call stack for exceptions?

### ISSUE THREAD

**Issue by ApsarasX** [bytecodealliance/wasm-micro-runtime#2214]
source code

```c
#include <assert.h>

int main() {
    assert(0);
}
```

compile instruction
```sh
wasi-sdk/bin/clang example.c -O0 -o example.wasm
```

wasmtime output
```
Assertion failed: 0 (x.c: main: 4)
Error: failed to run main module `x.wasm`

Caused by:
    0: failed to invoke command default
    1: error while executing at wasm backtrace:
           0:  0x248 - <unknown>!abort
           1:  0x291 - <unknown>!__assert_fail
           2:  0x1c0 - <unknown>!__original_main
           3:  0x15c - <unknown>!_start
           4: 0x3c1c - <unknown>!_start.command_export
    2: wasm trap: wasm `unreachable` instruction executed
```

wasmer output
```
Assertion failed: 0 (x.c: main: 4)
error: failed to run `x.wasm`
│   1: RuntimeError: unreachable
           at abort (<module>[14]:0x248)
           at __assert_fail (<module>[15]:0x291)
           at __original_main (<module>[5]:0x1c0)
           at _start (<module>[4]:0x15c)
           at _start.command_export (<module>[45]:0x3c1c)
╰─▶ 2: unreachable
```

But WAMR output
```
Assertion failed: 0 (x.c: main: 4)
Exception: unreachable
```

There is no exception call stack in the output of WAMR.

Do you have any plans to implement the feature of dumping the exception call stack in WAMR? 

Could you give a brief explanation of the implementation idea?

**Comment by TianlongLiang** [comment]
Hi, @ApsarasX , you can refer to [this document](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/build_wamr.md#enable-dump-call-stack-feature) or [this blog](https://bytecodealliance.github.io/wamr.dev/blog/understand-the-wamr-stacks/#execution-environment-exec_env)

TL;DR: The dumping option can be enabled through adding compilation option -DWAMR_BUILD_DUMP_CALL_STACK=1 for runtime, and --enable-dump-call-stack for the wamrc.
